### DISCUSSION TITLE
How to use avr-gdb with MK404

### DISCUSSION THREAD

**Discussion by GilesBathgate** [vintagepc/MK404#344] [General]
So I tried the instructions here: https://github.com/vintagepc/MK404/wiki/Debugging-Printer-Firmware-with-MK404

I have a file Firmware.ino.exe in `~/projects/Prusa-Firmware-build/Firmware.ino.elf`

I run: `/MK404 Prusa_MK2_mR13 -f ~/projects/PF-build-hex/FW-Build4481/BOARD_RAMBO_MINI_1_3/FW-Build4481-1_75mm_MK2-RAMBo13a-E3Dv6full-EN_ONLY.hex --gdb`

In a separate terminal I run:
```
$avr-gdb

(gdb) load ~/projects/Prusa-Firmware-build/Firmware.ino.elf
You can't do that when your target is `None'
(gdb) target extended-remote :1234
Remote debugging using :1234
warning: No executable has been specified and target does not support
determining executable automatically.  Try using the "file" command.
0x00000000 in ?? ()
(gdb)  load ~/projects/Prusa-Firmware-build/Firmware.ino.elf
Remote target does not support flash erase
```

I seem to be missing some Mad GDB skills :(

**Comment by GilesBathgate** [comment]
I think perhaps its supposed to be `file` not `load`

![image](https://user-images.githubusercontent.com/444143/144492709-0325adb5-0060-48e9-a88c-f5344af8fbc1.png)

**Comment by GilesBathgate** [comment]
@vintagepc Any chance you could update the wiki to say `file` not `load` ;)

**Comment by vintagepc** [comment]
Oops! yes, fixed! :)

**Comment by vintagepc** [comment]
btw, avr-insight is a nice GUI for debugging that I've used in the past.

**Reply by GilesBathgate** [reply]
Thanks ill give it a try. I did manage to get it to break in the suspect firmware code, no idea why its not moving the steppers still...
